src/wp-includes/class-wp-token-map.php

```diff
@@ -0,0 +1,728 @@
+<?php
+
+/**
+ * Class for efficiently looking up and mapping string keys to string values, with limits.
+ *
+ * @package    WordPress
+ * @since      6.6.0
+ */
+
+/**
+ * WP_Token_Map class.
+ *
+ * Use this class in specific circumstances with a static set of lookup keys which map to
+ * a static set of transformed values. For example, this class is used to map HTML named
+ * character references to their equivalent UTF-8 values.
+ *
+ * This class works differently than code calling `in_array()` and other methods. It
+ * internalizes lookup logic and provides helper interfaces to optimize lookup and
+ * transformation. It provides a method for precomputing the lookup tables and storing
+ * them as PHP source code.
+ *
+ * All tokens and substitutions must be shorter than 256 bytes.
+ *
+ * Example:
+ *
+ *     $smilies = WP_Token_Map::from_array( array(
+ *         '8O' => '😯',
+ *         ':(' => '🙁',
+ *         ':)' => '🙂',
+ *         ':?' => '😕',
+ *      ) );
+ *
+ *      true  === $smilies->contains( ':)' );
+ *      false === $smilies->contains( 'simile' );
+ *
+ *      '😕' === $smilies->read_token( 'Not sure :?.', 9, $bytes_skipped );
+ *      2    === $bytes_skipped;
+ *
+ *      echo $smilies->precomputed_php_source_table( '    ' );
+ *      // Output.
```

Comment:
Nit: it took me some time to understand what `echo $smilies->precomputed_php_source_table( '    ' );` does. Maybe the comment could get updated here and in other places to:
```suggestion
 *      // Output:
```

Reply:
I've expanded the comment and rearranged the example. Mind giving it another lookover?

Reply:
That is crystal clear now 👍